app/models/artefact.rb

```diff
@@ -344,7 +344,7 @@ def validate_redirect_url
 
   def valid_redirect_url_path?(target)
     URI.parse(target)
-    target.starts_with?("/") && target !~ %r{//} && target !~ %r{./\z}
+    target !~ %r{[^:]//}
```

Comment:
The `%r{[^:]//}` is just me making the existing `%r{//}` more specific so that it doesn't match "https://foo.gov.uk", but does match "/foo//bar". I'm guessing that it was intended to stop duplicate slashes being included.